Core/Src/main.c

```diff
@@ -87,18 +89,36 @@ int main(void)
   /* Initialize all configured peripherals */
   MX_GPIO_Init();
   MX_USART2_UART_Init();
+  MX_SPI1_Init();
+  MX_TIM1_Init();
   /* USER CODE BEGIN 2 */
 
   /* USER CODE END 2 */
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  HAL_TIM_PWM_Start(*htim, TIM_CHANNEL_1); //assuming channel 1 based on schematic "servo header". Timer needs to be started only once in program, so it is outside loop
   while (1)
   {
-    /* USER CODE END WHILE */
+	  int pTxData = 0;
+	  int pRxData = 0;
+	  HAL_Delay(10);
+	  *pTxData = 0b000000011000000000000000;
```

Comment:
I am not sure what you are trying to do here...

so it is true that the spi transfer function does take a pointer but it is a pointer to the array of data you want to send. Also remember, SPI can only send 1 byte at a time so that means you need to initialize an array to uint8_t.

Also remember how much data you are trying to send, 1, 2 or 3 bytes?

The same is said about the receive buffer too...